lib/rig_inbound_gateway_web/v1/longpolling_controller.ex

```diff
@@ -53,29 +54,40 @@ defmodule RigInboundGatewayWeb.V1.LongpollingController do
 
   # starts a new session
   defp process_request(true, conn) do
-    case Session.start(conn.query_params) do
-      {:ok, session_pid} ->
-        Logger.debug(fn ->
-          "new Longpolling connection (pid=#{inspect(session_pid)}, params=#{
-            inspect(conn.query_params)
-          })"
-        end)
-
-        conn
-        |> with_allow_origin()
-        |> put_resp_cookie("connection_token", session_pid |> Connection.Codec.serialize())
-        |> put_resp_cookie("last_event_id", Jason.encode!("first_event"))
-        |> put_resp_header("content-type", "application/json; charset=utf-8")
-        |> put_resp_header("cache-control", "no-cache")
-        |> put_status(200)
-        |> text(Jason.encode!("ok"))
-
+    with {:ok, _n_connections} <- ConnectionLimit.check_rate_limit(),
+         {:ok, session_pid} <- Session.start(conn.query_params) do
+      Logger.debug(fn ->
+        "new Longpolling connection (pid=#{inspect(session_pid)}, params=#{
+          inspect(conn.query_params)
+        })"
+      end)
+
+      conn
+      |> with_allow_origin()
+      |> put_resp_cookie("connection_token", session_pid |> Connection.Codec.serialize())
+      |> put_resp_cookie("x-last_event_id", Jason.encode!("first_event"))
+      |> put_resp_header("content-type", "application/json; charset=utf-8")
+      |> put_resp_header("cache-control", "no-cache")
+      |> put_status(200)
+      |> text("ok")
```

Comment:
hm wait, something's off here... the response type is json, but the response is _created_ using `text`, which means the body is actually not valid json... So _that_ is why there was this weird `text(Jason.encode!("ok"))` thing there in the first place! So I guess the right way to do it would be to
- use `json("ok")` here
- remove the `put_resp_header("content-type"` line above, as `json` already sets the content-type header anyway